great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -2776,25 +2785,27 @@ def _build_datasource_from_config(self, config: DatasourceConfig) -> Datasource:
         """
         # We convert from the type back to a dictionary for purposes of instantiation
         serializer = DictConfigSerializer(schema=datasourceConfigSchema)
-        config_dict: dict = serializer.serialize(config)
+        raw_config_dict: dict = serializer.serialize(raw_config)
+        substituted_config_dict: dict = serializer.serialize(substituted_config)
+        substituted_config_dict["raw_config"] = raw_config_dict
```

Comment:
Should `DatasourceConfig` and `DatasourceConfigSchema` be updated to reflect these new changes?

Reply:
Good question. I think so. It would have to be optional to not break existing code.